data_juicer/ops/mapper/mllm_mapper.py

```diff
@@ -0,0 +1,88 @@
+from typing import Dict
+
+import torch
+
+from data_juicer.ops.base_op import OPERATORS, Mapper
+from data_juicer.ops.op_fusion import LOADED_IMAGES
+from data_juicer.utils.mm_utils import load_image
+from data_juicer.utils.model_utils import get_model, prepare_model
+
+OP_NAME = 'mllm_mapper'
+torch.set_num_threads(1)
+
+
+@LOADED_IMAGES.register_module(OP_NAME)
+@OPERATORS.register_module(OP_NAME)
+class MllmMapper(Mapper):
+    """Mapper to optimize instruction.
+    Recommended model list: [
+        liuhaotia/llava-v1.6-vicuna-7b
+    ]
+    """
+    _accelerator = 'cuda'
+
+    def __init__(self,
+                 hf_model: str = 'liuhaotia/llava-v1.6-vicuna-7b',
+                 max_new_tokens=256,
+                 sampling_params: Dict = {},
+                 *args,
+                 **kwargs):
+        """
+        Initialization method.
+        :param hf_model: Hugginface model id.
+        :param max_new_tokens: the maximum number of new tokens
+            generated by the model.
+        :param sampling_params: Sampling hyperparameters for text generation.
+            e.g {'temperature': 0.9, 'top_p': 0.95}
+        :param args: extra args
+        :param kwargs: extra args
+        """
+        super().__init__(*args, num_proc=1, **kwargs)
+
+        self.hf_model = hf_model
+        self.model_key = prepare_model(model_type='huggingface',
+                                       pretrained_model_name_or_path=hf_model)
+        self.sampling_params = sampling_params
+        self.max_new_tokens = max_new_tokens
+
+    def process(self, sample=None, rank=None):
+
+        model, processor = get_model(self.model_key, rank=rank, use_cuda=True)
```

Comment:
The model should be loaded after preliminary checks to avoid unnecessary overhead.